### ISSUE TITLE
Ensure the built DLL's are used when running Cgreen's own tests

### ISSUE THREAD

**Issue by thoni56** [cgreen-devs/cgreen#8]
On DLL-platforms, in this case Cygwin, the dynamic libraries are not loaded in the same way as for *nix. Since the cygcgreen.dll is built in in the src directory and the test executable is run in the tests subtree, CGreen's own freshly built DLL is not used.

Instead you need to install *before* running the tests. This works but is obviously easy to forget (and has bitten me multiple times causing long frustrating debugging sessions before realizing that I recognized the problem...)

There are [indications](https://www.google.se/search?sourceid=chrome-psyapi2&ion=1&espv=2&ie=UTF-8&q=cmake%20find%20dll%20when%20running%20tests&oq=cmake%20find%20dll%20when%20running%20tests&aqs=chrome..69i57.10639j0j7) that it would be possible to fix this using some CMake magic.

**Comment by d-meiser** [comment]
I assume the CMake magic you're referring to is to add something like the following for each test `t`:
```
if (WIN32)
  set_tests_properties(${t} PROPERTIES ENVIRONMENT PATH=${PROJECT_SOURCE_DIR}/src)
endif ()
```

I use this regularly  for my own projects and can confirm that this works on cygwin.

**Comment by thoni56** [comment]
Fixed this with 057f9e2248f7af85109fbea95a05f5a176caebe5.
